x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -51,6 +52,11 @@ protected Object unwrapCustomValue(Object values) {
                 return parseDateString(values);
             }
         }
+        if (dataType == DATETIME_NANOS) {
+            if (values instanceof String) {
+                return DateUtils.asDateTimeWithNanos(values.toString(), zoneId());
```

Comment:
Is `.toString()` preferable to the cast?

Reply:
yep, generally, instanceof and casting should be avoided if possible.